cleanlab_studio/cli/classes/dataset.py

```diff
@@ -1,16 +1,45 @@
 from abc import abstractmethod
-from typing import Optional, List, Dict, Generator, Any
+import contextlib
+from typing import Optional, List, IO, Generator, Any, Iterator, Dict, Union
 
 import pandas as pd
 
 from cleanlab_studio.cli.types import RecordType
 
 
+FileObj = Union[IO[str], IO[bytes]]
+
+
 class Dataset:
-    def __init__(self, filepath: str):
-        self.filepath = filepath
+    READ_ARGS: Dict[str, str] = {}
+
+    def __init__(self, filepath: Optional[str] = None, fileobj: Optional[FileObj] = None):
+        if filepath is None and fileobj is None:
+            raise ValueError(
+                "One of `filepath` or `fileobj` must be provided to initialize `Dataset`."
+            )
+
+        self._filepath: Optional[str] = filepath
+        self._fileobj: Optional[FileObj] = fileobj
+
         self._num_rows: Optional[int] = None
 
+    @contextlib.contextmanager
+    def fileobj(self) -> Iterator[FileObj]:
+        """Yields open IO object to dataset file."""
+        if self._filepath is not None:
+            with open(self._filepath, **self.READ_ARGS) as dataset_file:  # type: ignore
+                yield dataset_file
+
+        elif self._fileobj is not None:
+            try:
```

Comment:
I think we have to reconsider how we deal with the file object case. Since we're using `with` statements in the Dataset methods, after calling one method, the file will be closed. So calling multiple methods on the same Dataset object will fail unless the caller reopens the file in between

Reply:
That is true when we use a context manager to create a file object -- the Dataset object created will only be valid while the file object is open. 

However, I believe we frequently have persistent file stream objects. See werkzeug.FileStorage for example -- that's how we receive filepond objects.